action/protocol/staking/protocol.go

```diff
@@ -214,6 +217,9 @@ func (p *Protocol) CreatePreStates(ctx context.Context, sm protocol.StateManager
 	bcCtx := protocol.MustGetBlockchainCtx(ctx)
 	blkCtx := protocol.MustGetBlockCtx(ctx)
 	hu := config.NewHeightUpgrade(&bcCtx.Genesis)
+	if p.enableArchiveMode && p.hu.IsPost(config.Fairbank, blkCtx.BlockHeight) && p.hu.IsPre(config.Greenland, blkCtx.BlockHeight) {
```

Comment:
not sure this or just `p.enableArchiveMode && p.hu.IsPost(config.Fairbank, blkCtx.BlockHeight)`?

Reply:
i think should be p.enableArchiveMode && p.hu.IsPre(Greenland)

we switch to nsV2 at fairbank height, but some buckets are already created before fairbank, and as long as there is bucket, we need to save staking protocol balance

Reply:
what about the balance after greenland?
I think it's p.enableArchiveMode && p.hu.IsPost(config.Fairbank-720, blkCtx.BlockHeight)